utils/dev-tool/src/server.rs

```diff
@@ -0,0 +1,86 @@
+use crate::utils::HashInfo;
+use crate::{panic_if_unsuccessful, utils, ToolEnvironment};
+
+use std::fs;
+use std::process::{Command, Stdio};
+
+pub fn build_server(tool_env: ToolEnvironment) {
+    dotenv::from_path(utils::local_env_path(&tool_env.root_dir)).unwrap();
+
+    let build_results = Command::new("cargo")
+        .args(["build", "-p", "lockbook-server"])
+        .status()
+        .unwrap();
+
+    panic_if_unsuccessful!(build_results);
+
+    let server_path = tool_env.target_dir.join("debug/lockbook-server");
+    let new_server_path =
+        server_path.with_file_name(format!("lockbook-server-{}", tool_env.commit_hash));
+
+    fs::rename(server_path, &new_server_path).unwrap();
+
+    let hash_info = HashInfo::new(&tool_env.hash_info_dir, &new_server_path, &tool_env.commit_hash);
+    hash_info.save();
+}
+
+pub fn run_server_detached(tool_env: ToolEnvironment) {
+    let port = port_scanner::request_open_port().unwrap();
+
+    dotenv::from_path(utils::local_env_path(&tool_env.root_dir)).unwrap();
+
+    let mut hash_info = HashInfo::get_from_dir(&tool_env.hash_info_dir, &tool_env.commit_hash);
+
+    Command::new(&hash_info.server_binary_path)
+        .env("SERVER_PORT", port.to_string())
+        .stderr(Stdio::null())
+        .stdout(Stdio::null())
+        .spawn()
+        .unwrap();
```

Comment:
in the future we want to wait 5s and see if the process is still running and then move on. Perhaps also we'd want to pipe logs somewhere.

Reply:
I ended up doing that since it was quick and easy.